### DISCUSSION TITLE
Will "Prequant BitsAndBytes models with TP" be supported?

### DISCUSSION THREAD

**Discussion by congwupiece** [vllm-project/vllm#10117] [Q&A]
when seve a BitsAndBytes  model with --tensor-parallel-size N
it come that
ValueError: Prequant BitsAndBytes models with TP is not supported.Please try with PP.

**Comment by aidando73** [comment]
Getting same error.

Btw, what is PP?

**Comment by jeejeelee** [comment]
@chenqianfzh  What difficulties are there with this support?
@aidando73  PP stands for pipeline parallelism

**Reply by chenqianfzh** [reply]
> @chenqianfzh What difficulties are there with this support? @aidando73 PP stands for pipeline parallelism

Michael and I had discussed about the lack of support of TP to prequant bnb models, in PR https://github.com/vllm-project/vllm/pull/8434.  We agreed PP is a reasonable choice than TP.

![image](https://github.com/user-attachments/assets/63348dfc-4732-4d4a-b971-523cbaef65fb)

**Comment by apohllo** [comment]
Could you please change the message? It's obvious to me what is tensor parallelizm, but seeing the message I had no idea was is PP vs. TP. This is such a minor change, but will save a lot of people I belive.

**Reply by jeejeelee** [reply]
> Could you please change the message? It's obvious to me what is tensor parallelizm, but seeing the message I had no idea was is PP vs. TP. This is such a minor change, but will save a lot of people I belive.

Make sense, I am handling this now

**Reply by jeejeelee** [reply]
@apohllo  See: https://github.com/vllm-project/vllm/pull/12320
